src/resolve/resolved-conf.c

```diff
@@ -376,6 +454,64 @@ int config_parse_dnssd_txt(const char *unit, const char *filename, unsigned line
         return 0;
 }
 
+
+int config_parse_dns_stub_listener_extra_address(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_free_ DNSStubListenerExtra *l = NULL;
+        _cleanup_free_ char *k = NULL;
+        Manager *m = userdata;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                m->dns_extra_stub_listeners = ordered_hashmap_free(m->dns_extra_stub_listeners);
+                return 0;
+        }
+
+        r = dns_stub_extra_new(&l);
+        if (r < 0)
+                return r;
+
+        r = socket_address_parse_ip(&l->address, rvalue);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse address in DnsStubListenExtra='%s', ignoring: %m", rvalue);
+                return 0;
+        }
+
+        if (socket_address_family(&l->address) == AF_INET && l->address.sockaddr.in.sin_port == 0)
+                l->address.sockaddr.in.sin_port = htobe16(53);
+        else if (l->address.sockaddr.in6.sin6_port == 0)
+                l->address.sockaddr.in6.sin6_port = htobe16(53);
+
+        r = ordered_hashmap_ensure_put(&m->dns_extra_stub_listeners, &dns_stub_listener_extra_address_hash_ops, l, l);
+        if (r < 0) {
+                if (r == -ENOMEM)
+                        return log_oom();
+
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to store DnsStubListenExtra='%s', ignoring assignment: %m", k ? k : rvalue);
+                return 0;
+        }
+
+        TAKE_PTR(l);
+
+        return 0;
+}
```

Comment:
hmm, to simplify the code I'd probably create one listener object per protocol. i.e. if an address for both udp/tcp is specified, then just allocate two listener objects here. That way each listener objects is exactly one fd, and things are pretty easy later on.